software/bsg_manycore_lib/bsg_manycore.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef _BSG_MANYCORE_HPP_
+#define _BSG_MANYCORE_HPP_
+
+#include <cstdint>
+#include <bsg_manycore_arch.h>
+
+
+
+template<typename T>
+T *bsg_remote_pointer(unsigned char x, unsigned char y, T* local_addr) {
```

Comment:
I think this is using tile-group EVA space? "bsg_remote_pointer" seems like a really vague name. maybe others can chime in. Also we could probably use this macro.
https://github.com/bespoke-silicon-group/bsg_manycore/blob/master/software/bsg_manycore_lib/bsg_manycore_arch.h#L68

Reply:
Yes but the whole point of adding templatized c++ barriers is to move away from macros, and get more visibility/type safetyness from c++ code. the bsg_remote_pointer is basically the templatized version of the bsg_remote_ptr macro. And I'm putting it in a separate file, so it gets populated over time with replacements for macros in bsg_mancyore.h

Reply:
I agree with Tommy on the naming. "Remote Pointer" is too generic, if this is only used for TG pointers bsg_tile_group_remote_pointer would be better, though verbose. It's also more accurate since bsg_tile_group_remote_ptr (the macro in the file above) takes a type like this does.

If you want backward compatibility, have bsg_remote_pointer use bsg_tile_group_remote pointer and mark bsg_remote_pointer as deprecated.

Reply:
I agree with both of you on the confusing name, it makes sense to rename it to bsg_tile_group_remote_pointer. But as far as backward compatibility goes, the compiler barks at me for deprecating a template function, plus bsg_remote_pointer is not used anywhere as just added it for the purposes of template barrier, so I just renamed it to bsg_tile_group_remote_pointer and removed the bsg_remote_pointer. Does that make sense?

Reply:
Yeah, makes sense.